diff --git a/src/x86/unix64.S b/src/x86/unix64.S
--- a/src/x86/unix64.S
+++ b/src/x86/unix64.S
@@ -873,6 +873,6 @@
 #endif
 
 #endif /* __x86_64__ */
-#if defined __ELF__ && (defined __linux__ || defined __FreeBSD__)
+#if defined __ELF__ && (defined __linux__ || defined __FreeBSD__ || defined __pedigree__)
 	.section	.note.GNU-stack,"",@progbits
 #endif
diff --git a/src/x86/win64.S b/src/x86/win64.S
--- a/src/x86/win64.S
+++ b/src/x86/win64.S
@@ -260,6 +260,6 @@
 #endif
 #endif /* __x86_64__ */
 
-#if defined __ELF__ && (defined __linux__ || defined __FreeBSD__)
+#if defined __ELF__ && (defined __linux__ || defined __FreeBSD__ || defined __pedigree__)
 	.section	.note.GNU-stack,"",@progbits
 #endif
