--- lua-5.1.4/Makefile	2008-08-12 10:40:48.000000000 +1000
+++ lua-5.1.4-new/Makefile	2010-09-25 21:05:05.976024186 +1000
@@ -9,16 +9,16 @@ PLAT= none
 
 # Where to install. The installation starts in the src and doc directories,
 # so take care if INSTALL_TOP is not an absolute path.
-INSTALL_TOP= /usr/local
-INSTALL_BIN= $(INSTALL_TOP)/bin
+INSTALL_TOP= 
+INSTALL_BIN= $(INSTALL_TOP)/applications
 INSTALL_INC= $(INSTALL_TOP)/include
-INSTALL_LIB= $(INSTALL_TOP)/lib
-INSTALL_MAN= $(INSTALL_TOP)/man/man1
+INSTALL_LIB= $(INSTALL_TOP)/libraries
+INSTALL_MAN= $(INSTALL_TOP)/support/lua/man
 #
 # You probably want to make INSTALL_LMOD and INSTALL_CMOD consistent with
 # LUA_ROOT, LUA_LDIR, and LUA_CDIR in luaconf.h (and also with etc/lua.pc).
-INSTALL_LMOD= $(INSTALL_TOP)/share/lua/$V
-INSTALL_CMOD= $(INSTALL_TOP)/lib/lua/$V
+INSTALL_LMOD= $(INSTALL_TOP)/support/lua/$V
+INSTALL_CMOD= $(INSTALL_LIB)/lua/$V
 
 # How to install. If your install program does not support "-p", then you
 # may have to run ranlib on the installed liblua.a (do "make ranlib").
