--- a/libstdc++-v3/src/c++17/fs_ops.cc
+++ b/libstdc++-v3/src/c++17/fs_ops.cc
@@ -50,6 +50,9 @@
 #ifdef _GLIBCXX_HAVE_SYS_STATVFS_H
 # include <sys/statvfs.h> // statvfs
 #endif
+#ifdef __pedigree__
+# include <sys/time.h> // timeval, utimes
+#endif
 #if !_GLIBCXX_USE_UTIMENSAT && _GLIBCXX_HAVE_UTIME_H
 # include <utime.h> // utime
 #endif
@@ -1083,7 +1086,29 @@
 {
   auto d = internal_file_clock::_S_to_sys(new_time).time_since_epoch();
   auto s = chrono::duration_cast<chrono::seconds>(d);
-#if _GLIBCXX_USE_UTIMENSAT
+#ifdef __pedigree__
+  auto us = chrono::duration_cast<chrono::microseconds>(d - s);
+  if (us < us.zero())
+    {
+      --s;
+      us += chrono::seconds(1);
+    }
+  posix::stat_type st;
+  if (posix::stat(p.c_str(), &st))
+    {
+      ec.assign(errno, std::generic_category());
+      return;
+    }
+  struct ::timeval times[2];
+  times[0].tv_sec = st.st_atim.tv_sec;
+  times[0].tv_usec = st.st_atim.tv_nsec / 1000;
+  times[1].tv_sec = static_cast<std::time_t>(s.count());
+  times[1].tv_usec = us.count();
+  if (::utimes(p.c_str(), times))
+    ec.assign(errno, std::generic_category());
+  else
+    ec.clear();
+#elif _GLIBCXX_USE_UTIMENSAT
   auto ns = chrono::duration_cast<chrono::nanoseconds>(d - s);
   if (ns < ns.zero()) // tv_nsec must be non-negative and less than 10e9.
     {
@@ -1470,10 +1495,29 @@
 {
   if (size > static_cast<uintmax_t>(std::numeric_limits<posix::off_t>::max()))
     ec.assign(EINVAL, std::generic_category());
+#ifdef __pedigree__
+  else
+    {
+      const int fd = posix::open(p.c_str(), O_WRONLY);
+      if (fd == -1)
+	ec.assign(errno, std::generic_category());
+      else
+	{
+	  const int result = ::ftruncate(fd, size);
+	  const int error = errno;
+	  posix::close(fd);
+	  if (result)
+	    ec.assign(error, std::generic_category());
+	  else
+	    ec.clear();
+	}
+    }
+#else
   else if (posix::truncate(p.c_str(), size))
     ec.assign(errno, std::generic_category());
   else
     ec.clear();
+#endif
 }
 
 
