diff --git a/Utilities/cmlibuv/src/unix/core.c b/Utilities/cmlibuv/src/unix/core.c
--- a/Utilities/cmlibuv/src/unix/core.c
+++ b/Utilities/cmlibuv/src/unix/core.c
@@ -1099,6 +1099,11 @@


 static int uv__getrusage(int who, uv_rusage_t* rusage) {
+#if defined(__pedigree__)
+  (void) who;
+  (void) rusage;
+  return UV_ENOSYS;
+#else
   struct rusage usage;

   if (getrusage(who, &usage))
@@ -1137,6 +1142,7 @@
 #endif

   return 0;
+#endif
 }


@@ -1731,6 +1737,11 @@
  * So the output parameter priority is actually the nice value.
 */
 int uv_thread_getpriority(uv_thread_t tid, int* priority) {
+#if defined(__pedigree__)
+  (void) tid;
+  (void) priority;
+  return UV_ENOSYS;
+#else
   int r;
   int policy;
   struct sched_param param;
@@ -1758,6 +1769,7 @@

   *priority = param.sched_priority;
   return 0;
+#endif
 }
 #endif

@@ -1784,7 +1796,11 @@
  * If the function fails, the return value is non-zero.
 */
 int uv_thread_setpriority(uv_thread_t tid, int priority) {
-#if !defined(__GNU__)
+#if defined(__pedigree__)
+  (void) tid;
+  (void) priority;
+  return UV_ENOSYS;
+#elif !defined(__GNU__)
   int r;
   int min;
   int max;
diff --git a/Utilities/cmlibuv/src/unix/fs.c b/Utilities/cmlibuv/src/unix/fs.c
--- a/Utilities/cmlibuv/src/unix/fs.c
+++ b/Utilities/cmlibuv/src/unix/fs.c
@@ -85,6 +85,32 @@

 #if defined(_AIX) && _XOPEN_SOURCE <= 600
 extern char *mkdtemp(char *template); /* See issue #740 on AIX < 7 */
+#endif
+
+#if defined(__pedigree__)
+static ssize_t uv__pread(int fd, void* buf, size_t count, off_t offset) {
+  (void) fd;
+  (void) buf;
+  (void) count;
+  (void) offset;
+  errno = ENOSYS;
+  return -1;
+}
+
+static ssize_t uv__pwrite(int fd,
+                          const void* buf,
+                          size_t count,
+                          off_t offset) {
+  (void) fd;
+  (void) buf;
+  (void) count;
+  (void) offset;
+  errno = ENOSYS;
+  return -1;
+}
+#else
+# define uv__pread pread
+# define uv__pwrite pwrite
 #endif

 #define INIT(subtype)                                                         \
@@ -433,9 +459,9 @@

     do
       if (is_pread)
-        r = pread(fd, p, n, off);
+        r = uv__pread(fd, p, n, off);
       else
-        r = pwrite(fd, p, n, off);
+        r = uv__pwrite(fd, p, n, off);
     while (r == -1 && errno == EINTR);

     if (r == -1) {
@@ -554,7 +580,7 @@
       r = readv(fd, bufs, nbufs);
   } else {
     if (nbufs == 1)
-      r = pread(fd, bufs->iov_base, bufs->iov_len, off);
+      r = uv__pread(fd, bufs->iov_base, bufs->iov_len, off);
     else if (nbufs > 1)
       r = uv__preadv(fd, bufs, nbufs, off);
   }
@@ -907,7 +933,7 @@

     do
       if (use_pread)
-        nread = pread(in_fd, buf, buflen, offset);
+        nread = uv__pread(in_fd, buf, buflen, offset);
       else
         nread = read(in_fd, buf, buflen);
     while (nread == -1 && errno == EINTR);
@@ -1243,7 +1269,7 @@
       r = writev(fd, bufs, nbufs);
   } else {
     if (nbufs == 1)
-      r = pwrite(fd, bufs->iov_base, bufs->iov_len, off);
+      r = uv__pwrite(fd, bufs->iov_base, bufs->iov_len, off);
     else if (nbufs > 1)
       r = uv__pwritev(fd, bufs, nbufs, off);
   }
@@ -1253,6 +1279,11 @@


 static ssize_t uv__fs_copyfile(uv_fs_t* req) {
+#if defined(__pedigree__)
+  (void) req;
+  errno = ENOSYS;
+  return -1;
+#else
   uv_fs_t fs_req;
   uv_file srcfd;
   uv_file dstfd;
@@ -1479,6 +1510,7 @@

   errno = UV__ERR(result);
   return -1;
+#endif
 }

 static void uv__to_stat(struct stat* src, uv_stat_t* dst) {
@@ -1752,7 +1784,11 @@
     X(COPYFILE, uv__fs_copyfile(req));
     X(FCHMOD, fchmod(req->file, req->mode));
     X(FCHOWN, fchown(req->file, req->uid, req->gid));
+#if defined(__pedigree__)
+    X(LCHOWN, (errno = ENOSYS, -1));
+#else
     X(LCHOWN, lchown(req->path, req->uid, req->gid));
+#endif
     X(FDATASYNC, uv__fs_fdatasync(req));
     X(FSTAT, uv__fs_fstat(req->file, &req->statbuf));
     X(FSYNC, uv__fs_fsync(req));
diff --git a/Utilities/cmlibuv/src/unix/pedigree.c b/Utilities/cmlibuv/src/unix/pedigree.c
--- a/Utilities/cmlibuv/src/unix/pedigree.c
+++ b/Utilities/cmlibuv/src/unix/pedigree.c
@@ -4,17 +4,11 @@
 #include "internal.h"

 #include <errno.h>
-#include <sys/resource.h>
 #include <time.h>

 int uv_resident_set_memory(size_t* rss) {
-  struct rusage usage;
-
-  if (getrusage(RUSAGE_SELF, &usage) < 0)
-    return UV__ERR(errno);
-
-  *rss = usage.ru_maxrss * 1024;
-  return 0;
+  (void) rss;
+  return UV_ENOSYS;
 }

 int uv_uptime(double* uptime) {
diff --git a/Source/kwsys/ProcessUNIX.c b/Source/kwsys/ProcessUNIX.c
--- a/Source/kwsys/ProcessUNIX.c
+++ b/Source/kwsys/ProcessUNIX.c
@@ -2911,7 +2911,11 @@
         sigaddset(&unblockSet, signum);
         sigprocmask(SIG_UNBLOCK, &unblockSet, 0);
         /* Raise the signal again.  */
+#if defined(__pedigree__)
+        kill(getpid(), signum);
+#else
         raise(signum);
+#endif
         /* We shouldn't get here... but if we do... */
         _exit(1);
       }
