--- prboom-2.5.0/configure	2010-08-20 18:36:51.712369962 +1000
+++ prboom-2.5.0-new/configure	2010-08-20 18:34:44.872371000 +1000
@@ -6264,6 +6264,8 @@ echo $ECHO_N "checking for SDL - version
       ac_save_LIBS="$LIBS"
+      SDL_CFLAGS=`echo $SDL_CFLAGS | sed "s/-I.*[ ]\+//g"`
       CFLAGS="$CFLAGS $SDL_CFLAGS"
       CXXFLAGS="$CXXFLAGS $SDL_CFLAGS"
+      SDL_LIBS=`echo $SDL_LIBS | sed "s/-L.*[ ]\+//g"`
       LIBS="$LIBS $SDL_LIBS"
       rm -f conf.sdltest
       if test "$cross_compiling" = yes; then
--- prboom-2.5.0/configure	2010-08-21 14:53:51.839734253 +1000
+++ prboom-2.5.0-new/configure	2010-08-21 14:53:25.758483000 +1000
@@ -5155,6 +5155,11 @@ fi
 rm -f conftest*
 
 fi
+
+if test "$cross_compiling"=xyes; then
+  ac_cv_type_uid_t=yes
+fi
+
 { echo "$as_me:$LINENO: result: $ac_cv_type_uid_t" >&5
 echo "${ECHO_T}$ac_cv_type_uid_t" >&6; }
 if test $ac_cv_type_uid_t = no; then
