--- examples/loadables/sync.c
+++ examples/loadables/sync.c
@@ -48,8 +48,15 @@ sync_builtin (WORD_LIST *list)
 
   if (list == 0)
     {  
+#ifdef __pedigree__
+      errno = ENOSYS;
+      builtin_error ("global synchronization is not supported: %s",
+                     strerror (errno));
+      return (EXECUTION_FAILURE);
+#else
       sync();
       return (EXECUTION_SUCCESS);
+#endif
     }
 
   status = EXECUTION_SUCCESS;
