--- a/crypto/bio/b_sock.c.old	2013-05-11 21:33:27.077135511 +1000
+++ b/crypto/bio/b_sock.c	2013-05-11 21:33:55.620298688 +1000
@@ -62,7 +62,7 @@
 #define USE_SOCKETS
 #include "cryptlib.h"
 #include <openssl/bio.h>
-#if defined(OPENSSL_SYS_NETWARE) && defined(NETWARE_BSDSOCK)
+#if (defined(OPENSSL_SYS_NETWARE) && defined(NETWARE_BSDSOCK)) || defined(__PEDIGREE__)
 #include <netdb.h>
 #if defined(NETWARE_CLIB)
 #include <sys/ioctl.h>
