--- a/crypto/dso/dso_dlfcn.c.old	2013-05-11 21:04:39.865500158 +1000
+++ b/crypto/dso/dso_dlfcn.c	2013-05-11 21:06:07.318781326 +1000
@@ -87,7 +87,7 @@ DSO_METHOD *DSO_METHOD_dlfcn(void)
      defined(__SCO_VERSION__) || defined(_SCO_ELF) || \
      (defined(__osf__) && !defined(RTLD_NEXT))     || \
      (defined(__OpenBSD__) && !defined(RTLD_SELF)) || \
-	defined(__ANDROID__)
+	defined(__ANDROID__) || defined(__PEDIGREE__)
 #  undef HAVE_DLINFO
 # endif
 #endif
