--- qemu-old/configure	2010-08-28 14:39:10.000000000 +1000
+++ qemu/configure	2010-08-28 14:39:32.000000000 +1000
@@ -93,13 +93,13 @@ ld="${cross_prefix}${ld}"
 
 # default flags for all hosts
 QEMU_CFLAGS="-fno-strict-aliasing $QEMU_CFLAGS"
-CFLAGS="-g $CFLAGS"
+CFLAGS="$CFLAGS"
 QEMU_CFLAGS="-Wall -Wundef -Wendif-labels -Wwrite-strings -Wmissing-prototypes $QEMU_CFLAGS"
 QEMU_CFLAGS="-Wstrict-prototypes -Wredundant-decls $QEMU_CFLAGS"
 QEMU_CFLAGS="-D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE $QEMU_CFLAGS"
 QEMU_CFLAGS="-U_FORTIFY_SOURCE $QEMU_CFLAGS"
 QEMU_CFLAGS="-I. -I\$(SRC_PATH) $QEMU_CFLAGS"
-LDFLAGS="-g $LDFLAGS"
+LDFLAGS="$LDFLAGS"
 
 gcc_flags="-Wold-style-declaration -Wold-style-definition"
 cat > $TMPC << EOF
