--- libtool-2.4-old/libltdl/m4/libtool.m4	2010-09-22 18:41:19.000000000 +1000
+++ libtool-2.4/libltdl/m4/libtool.m4	2010-11-16 09:42:42.000000000 +1000
@@ -2708,6 +2708,22 @@ osf3* | osf4* | osf5*)
   sys_lib_dlsearch_path_spec="$sys_lib_search_path_spec"
   ;;
 
+pedigree*)
+  version_type=linux
+  need_lib_prefix=yes
+  need_version=no
+  library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
+  soname_spec='${libname}${release}${shared_ext}$major'
+  finish_cmds=''
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+
+  hardcode_into_libs=no
+
+  dynamic_linker='Pedigree dylink'
+  ;;
+
+
 rdos*)
   dynamic_linker=no
   ;;
@@ -3260,6 +3276,10 @@ osf3* | osf4* | osf5*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+pedigree*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 rdos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -3887,6 +3907,9 @@ m4_if([$1], [CXX], [
 	;;
       esac
       ;;
+    pedigree*)
+      _LT_TAGVAR(lt_prog_compiler_pic, $1)='-fPIC'
+      ;;
     *qnx* | *nto*)
       # QNX uses GNU C++, but need to define -shared option too, otherwise
       # it will coredump.
@@ -4071,6 +4094,11 @@ m4_if([$1], [CXX], [
 	;;
       psos*)
 	;;
+      pedigree*)
+	_LT_TAGVAR(lt_prog_compiler_wl, $1)='-Wl,'
+	_LT_TAGVAR(lt_prog_compiler_pic, $1)='-fPIC'
+	_LT_TAGVAR(lt_prog_compiler_static, $1)='-static'
+        ;;
       solaris*)
 	case $cc_basename in
 	  CC* | sunCC*)
@@ -4215,6 +4243,10 @@ m4_if([$1], [CXX], [
       _LT_TAGVAR(lt_prog_compiler_pic, $1)='-fPIC -shared'
       ;;
 
+    pedigree*)
+      _LT_TAGVAR(lt_prog_compiler_pic, $1)='-fPIC'
+      ;;
+
     sysv4*MP*)
       if test -d /usr/nec; then
 	_LT_TAGVAR(lt_prog_compiler_pic, $1)=-Kconform_pic
@@ -4355,6 +4387,10 @@ m4_if([$1], [CXX], [
       _LT_TAGVAR(lt_prog_compiler_static, $1)='-non_shared'
       ;;
 
+    pedigree*)
+      _LT_TAGVAR(lt_prog_compiler_pic, $1)='-fPIC'
+      ;;
+
     rdos*)
       _LT_TAGVAR(lt_prog_compiler_static, $1)='-non_shared'
       ;;
@@ -4794,6 +4830,10 @@ _LT_EOF
       fi
       ;;
 
+    pedigree*)
+      _LT_TAGVAR(archive_cmds, $1)='$CC -shared $pic_flag $libobjs $deplibs $compiler_flags ${wl}-shared ${wl}-soname $wl$soname -o $lib'
+      ;;
+
     solaris*)
       if $LD -v 2>&1 | $GREP 'BFD 2\.8' > /dev/null; then
 	_LT_TAGVAR(ld_shlibs, $1)=no
@@ -6604,6 +6644,12 @@ if test "$_lt_caught_CXX_error" != yes; 
         esac
         ;;
 
+      pedigree*)
+	_LT_TAGVAR(hardcode_direct, $1)=no
+	_LT_TAGVAR(hardcode_shlibpath_var, $1)=no
+	_LT_TAGVAR(hardcode_direct_absolute, $1)=no
+        ;;
+
       psos*)
         # FIXME: insert proper C++ library support
         _LT_TAGVAR(ld_shlibs, $1)=no
