--- a/pp_sys.c
+++ b/pp_sys.c
@@ -2542,12 +2542,17 @@
         else {
             const char * const name = SvPV_nomg_const_nolen(sv);
             TAINT_PROPER("truncate");
-#ifdef HAS_TRUNCATE
+#if defined(HAS_TRUNCATE) && !defined(__pedigree__)
             if (truncate(name, len) < 0)
                 result = 0;
 #else
             {
+#ifdef __pedigree__
+                /* Pedigree translates ftruncate(), not truncate(). */
+                int mode = O_WRONLY;
+#else
                 int mode = O_RDWR;
+#endif
                 int tmpfd;
 
 #if defined(USE_64_BIT_RAWIO) && defined(O_LARGEFILE)
