--- a/meson.options
+++ b/meson.options
@@ -154,6 +154,6 @@
 
 option('file_monitor_backend',
        type : 'combo',
-       choices : ['auto', 'inotify', 'kqueue', 'libinotify-kqueue', 'win32'],
+       choices : ['auto', 'none', 'inotify', 'kqueue', 'libinotify-kqueue', 'win32'],
        value : 'auto',
        description : 'The name of the system API to use as a GFileMonitor backend')
--- a/meson.build
+++ b/meson.build
@@ -832,10 +833,21 @@
 # SVR4 sysinfo() does not behave the same as sysinfo() on other Unixes
 if host_system != 'sunos'
   functions += ['sysinfo']
+endif
+
+pedigree_unavailable_functions = []
+if host_system == 'pedigree'
+  pedigree_unavailable_functions = [
+    'close_range', 'sendmmsg'
+  ]
 endif
 
 foreach f : functions
-  if cc.has_function(f)
+  if f not in pedigree_unavailable_functions and cc.has_function(f)
     define = 'HAVE_' + f.underscorify().to_upper()
     glib_conf.set(define, 1)
     glib_conf_prefix = glib_conf_prefix + '#define @0@ 1\n'.format(define)
@@ -1088,7 +1101,7 @@
 endif
 
 # Check for pidfd_open(2)
-if cc.links('''#include <sys/syscall.h>
+if host_system != 'pedigree' and cc.links('''#include <sys/syscall.h>
                #include <sys/wait.h>
                #include <linux/wait.h>
                #include <unistd.h>
--- a/glib/glib-unix.c
+++ b/glib/glib-unix.c
@@ -996,6 +996,10 @@
     }
 
   return g_strdup (file_path);
+#elif defined (__pedigree__)
+  g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_NOSYS,
+               "g_unix_fd_query_path() not supported on Pedigree");
+  return NULL;
 #elif defined (__GNU__)
   /*
    * Hurd allows to open("/dev/fd/%u") to open the very same fd, but it's not
