--- a/shell.c
+++ b/shell.c
@@ -24761,13 +24761,16 @@
 #if !defined(_WIN32) && !defined(WIN32) && !defined(__minux)
 #include <sys/time.h>
 #include <sys/resource.h>
-/* VxWorks does not support getrusage() as far as we can determine */
+/* VxWorks does not support getrusage() as far as we can determine. */
 #if defined(_WRS_KERNEL) || defined(__RTP__)
 struct rusage {
   struct timeval ru_utime; /* user CPU time used */
   struct timeval ru_stime; /* system CPU time used */
 };
 #define getrusage(A,B) memset(B,0,sizeof(*B))
+/* Pedigree's libc wrapper uses a syscall the kernel does not implement. */
+#elif defined(__pedigree__)
+#define getrusage(A,B) memset(B,0,sizeof(*B))
 #endif
 
 /* Saved resource information for the beginning of an operation */
@@ -36579,6 +36582,8 @@
     }else{
 #if defined(_WIN32) || defined(WIN32)
       DebugBreak();
+#elif defined(__pedigree__) && defined(SIGTRAP)
+      kill(GETPID(), SIGTRAP);
 #elif defined(SIGTRAP)
       raise(SIGTRAP);
 #endif
